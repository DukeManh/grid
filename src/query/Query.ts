import { IQueryAction, QueryAction } from './QueryAction';

interface IQuery {
  from: (table: string, schema?: string) => IQueryAction;
}

export class Query implements IQuery {
  from(table: string, schema?: string) {
    return new QueryAction({ name: table, schema: schema ?? 'public' });
  }
}
