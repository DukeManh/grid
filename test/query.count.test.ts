import Query from '../src/query';

const query = new Query();

test('basic count table', async () => {
  const res = query.from('users').count().toSql();
  expect(res).toMatchSnapshot();
});
test('basic count table with filter', async () => {
  const res = query
    .from('users')
    .count()
    .filter('name', 'in', ['hello world', 'xin chao'])
    .toSql();
  expect(res).toMatchSnapshot();
});
