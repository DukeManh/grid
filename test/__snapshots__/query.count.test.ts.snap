// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic count table 1`] = `"select count(*) from public.users;"`;

exports[`count table with filter 1`] = `"select count(*) from public.users where name in ('hello world','xin chao');"`;

exports[`count table with multi filters 1`] = `"select count(*) from public.users where name in ('hello world','xin chao') and age > 18 and birth_year >= 2000;"`;
